File: tools/bench/answers/go_sonnet/s1/05_dedup.go
package main

import "fmt"

func main() {
	nums := []int{3, 1, 3, 2, 1, 4}
	seen := make(map[int]bool)
	result := []int{}
	for _, n := range nums {
		if !seen[n] {
			seen[n] = true
			result = append(result, n)
		}
	}
	for i, n := range result {
		if i > 0 {
			fmt.Print(" ")
		}
		fmt.Print(n)
	}
	fmt.Println()
}

File: tools/bench/answers/rust_haiku/s2/05_dedup.rs
fn main() {
    let nums = [3, 1, 3, 2, 1, 4];
    let mut seen = Vec::new();
    let mut result = Vec::new();

    for &n in &nums {
        if !seen.contains(&n) {
            seen.push(n);
            result.push(n);
        }
    }

    let output = result
        .iter()
        .map(|n| n.to_string())
        .collect::<Vec<_>>()
        .join(" ");
    println!("{}", output);
}
